src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AvoidDoubleBraceInitializationCheck.java

```diff
@@ -0,0 +1,136 @@
+ * </p>
+ * <p>
+ * Rationale: Double brace initialization may look cool, but it is considered
+ * as anti-pattern and should be avoided.
+ * This is also can lead to a hard-to-detect memory leaks, since created anonymous class
+ * is not static, it holds an implicit reference to the outer class instance.
+ * See this
+ * <a href="https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/">
+ * blog post</a> and
+ * <a href="https://www.baeldung.com/java-double-brace-initialization">
+ * article</a> for more details.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AvoidDoubleBraceInitialization&quot;/&gt;
+ * </pre>
+ * <p>
+ * Which results in the following violations:
+ * </p>
+ * <pre>
+ * class MyClass {
+ *   List list1 = new ArrayList&lt;Object&gt; { // violation
+ *     {
+ *       add(new Object());
+ *     }
+ *   };
+ *   List list2 = new ArrayList&lt;Object&gt; { // violation, comments are alone commas are ignored
+ *     ;
+ *     // my comment
+ *     {
+ *       add(new Object());
+ *     }
+ *   };
+ *   List list3 = new ArrayList&lt;Object&gt; { // OK, there is an additional field
+ *     private int field;
+ *     {
+ *       add(new Object());
+ *     }
+ *   };
+ * }
+ * </pre>
+ *
+ * @since 8.30
+ */
+@StatelessCheck
+public class AvoidDoubleBraceInitializationCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "avoid.double.brace.init";
+
+    /**
+     * List of token types that are used in {@link #hasOnlyInitialization(DetailAST)}.
+     */
+    private static final List<Integer> IGNORED_TYPES = Arrays.asList(
+        TokenTypes.INSTANCE_INIT,
+        TokenTypes.SEMI,
+        TokenTypes.LCURLY,
+        TokenTypes.RCURLY
+    );
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.OBJBLOCK};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getParent().getType() == TokenTypes.LITERAL_NEW
+            && hasOnlyInitialization(ast)) {
+            log(ast, MSG_KEY);
+        }
+    }
+
+    /**
+     * Checks that block has at least one instance init block and no other class members.
+     *
+     * @param objBlock token to check
+     * @return true if there is least one instance init block and no other class members,
+     *     false otherwise
+     */
+    private static boolean hasOnlyInitialization(DetailAST objBlock) {
+        final boolean hasInitBlock = objBlock.findFirstToken(TokenTypes.INSTANCE_INIT) != null;
+        final Predicate<DetailAST> hasMembers = token -> !IGNORED_TYPES.contains(token.getType());
```

Comment:
does it make sense to move this lambda to field or make it as method to use method reference ?
we do not have that much methods in class, so it will looks good to be not lambda, .... no need to instantiate it each time.

Reply:
I extracted lambda.

I couldn't find a readable way to extract method reference